scripts/upgrade_schema/upgrade_schema-59.X

```diff
@@ -0,0 +1,57 @@
+CREATE TABLE `seq_component` (
+  `id_seq_component`    BIGINT(20) UNSIGNED NOT NULL AUTO_INCREMENT COMMENT 'Auto-generated primary key',
+  `id_run`              BIGINT(20) UNSIGNED NOT NULL COMMENT 'Numeric run id',
+  `position`            TINYINT(1) UNSIGNED NOT NULL COMMENT 'Numeric position (lane number)',
+  `tag_index`           MEDIUMINT UNSIGNED  DEFAULT NULL COMMENT 'An optional numeric tag index',
+  `subset`              VARCHAR(10)         DEFAULT NULL COMMENT 'An optional sequence subset',
+  `digest`              CHAR(64) NOT NULL COMMENT 'A SHA256 hex digest of the component JSON representation as defined in npg_tracking::glossary::composition::component',
+  PRIMARY KEY (`id_seq_component`),
+  UNIQUE KEY `unq_seq_compon_d` (`digest`),
+  INDEX `unq_seq_compos_rp` (`id_run`, `position`, `tag_index`, `subset`)
+) ENGINE=InnoDB  DEFAULT CHARSET=latin1;
+
+CREATE TABLE `seq_composition` (
+  `id_seq_composition` BIGINT(20) UNSIGNED NOT NULL AUTO_INCREMENT COMMENT 'Auto-generated primary key',
+  `digest`             CHAR(64) NOT NULL COMMENT 'A SHA256 hex digest of the JSON representation of the composition as defined in npg_tracking::glossary::composition',
+  `size`               TINYINT UNSIGNED NOT NULL COMMENT 'Total number of components in a composition',
+  PRIMARY KEY (`id_seq_composition`),
+  UNIQUE KEY `unq_seq_compos_d` (`digest`),
+  UNIQUE KEY `unq_seq_compos_ps` (`id_seq_composition`, `size`) /* To allow for a composite foreign key in seq_component_composition */
+) ENGINE=InnoDB  DEFAULT CHARSET=latin1;
+
+CREATE TABLE `seq_component_composition` (
+  `id_seq_comcom`      BIGINT(20) UNSIGNED NOT NULL AUTO_INCREMENT COMMENT 'Auto-generated primary key',
+  `id_seq_component`   BIGINT(20) UNSIGNED NOT NULL COMMENT 'A foreign key referencing the id_seq_component column of the seq_component table',
+  `id_seq_composition` BIGINT(20) UNSIGNED NOT NULL COMMENT 'A foreign key referencing the id_seq_composition column of the seq_composition table',
+  `size`               TINYINT UNSIGNED NOT NULL    COMMENT 'Total number of components in a composition',
+  PRIMARY KEY (`id_seq_comcom`),
+  UNIQUE KEY `unq_seq_comcom` (`id_seq_component`, `id_seq_composition`, `size`),
```

Comment:
> Should size be in this unique key?
No. Thank you!
Scenario - Inserting a new record into the linking table.

    Without 'size' in the key a duplicate relationship will be rejected
    straight away. With 'size' in the key a duplicate relationship with
    existing size will be rejected, a duplicate relationship with a
    different size will be OK, which is nonsense. In practice the insert
    will be rejected because it will violate the composite foreign key
    constraint, but it's better not to have 'size' in that unique key.









-- 
 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.